File: src/main/java/me/nasukhov/TukitaLearner/dictionary/ImportDictionary.java
package me.nasukhov.TukitaLearner.dictionary;

import com.opencsv.CSVReader;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class ImportDictionary {
    private final DictionaryRepository storage;

    public ImportDictionary(DictionaryRepository dictionaryRepository) {
        storage = dictionaryRepository;
    }

    public void run() {
        ClassLoader classloader = Thread.currentThread().getContextClassLoader();
        InputStream is = classloader.getResourceAsStream("dictionary.csv");
        if (is == null) {
            throw new RuntimeException("Could not load dictionary resource");
        }

        boolean headline = true;
        List<Word> words = new ArrayList<>();
        try (CSVReader csvReader = new CSVReader(new InputStreamReader(is))) {
            String[] columns = null;
            while ((columns = csvReader.readNext()) != null) {
                // skipping columns definition row
                if (headline) {
                    headline = false;
                    continue;
                }

                Word word = new Word(Word.canonize(columns[3]), Word.canonize(columns[1]), columns[13], PartOfSpeech.fromValue(columns[8]));
                words.add(word);
            }

            storage.add(words);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: src/main/kotlin/me/nasukhov/TukitaLearner/dictionary/ImportDictionary.kt
package me.nasukhov.TukitaLearner.dictionary

import com.opencsv.CSVReader
import org.springframework.context.annotation.Lazy
import org.springframework.stereotype.Component
import java.io.InputStreamReader

@Component
@Lazy
class ImportDictionary(private val storage: DictionaryRepository) {
    fun run() {
        val classloader = Thread.currentThread().getContextClassLoader()
        val loader = classloader.getResourceAsStream("dictionary.csv")
        requireNotNull(loader) { "Dictionary file is missing" }

        var headline = true
        val words = mutableListOf<Word>()

        CSVReader(InputStreamReader(loader)).use {
            it.forEach { row ->
                // skipping columns definition row
                if (headline) {
                    headline = false
                    return@forEach
                }

                val word = Word(
                    word = Word.canonize(row[3]),
                    translation = Word.canonize(row[1]),
                    description = row[13],
                    partOfSpeech = PartOfSpeech.fromValue(row[8])
                )
                words.add(word)
            }

            storage.saveAllAndFlush(words)
        }

    }
}
